### PR TITLE
Improve documentation a bit.

### PR DESCRIPTION
**Pull request by hboetes**
- Since dependencies are important to both people who want to run from git and packagers I created DEPENDENCIES.md and I move various dependencies mentions in other files to this one, replacing them with links to the new file
- Move licence next to legal stuff
- Improve the descriptions a bit
- Improve wording a bit
- Improve the layout a bit

### CODE REVIEW

**Comment by hboetes** [comment]
I suppose I need to squash all these commits?

**Comment by mathiascode** [comment]
We can probably squash-merge instead.

**Comment by hboetes** [comment]
Cool, thanks.

**Comment by kiplingw** [comment]
@mathiascode, please bump the version to 2.0.1 so unstable PPA's package will be more easily distinguished from the stable release we put out yesterday.

**Comment by mathiascode** [comment]
@kiplingw Done in https://github.com/Nicotine-Plus/nicotine-plus/pull/394, switched to a dev tag to distinguish from stable.

File: doc/DEPENDENCIES.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+* On Redhat/Fedora based distributions:
+```
+sudo dnf install gobject-introspection gtk3 python3-dbus python3-gobject python3-mutagen
+```
+* On Debian/Ubuntu based distributions:
+```
+sudo apt install gobject-introspection gir1.2-gtk-3.0 python3-dbus python3-gi python3-mutagen
+```
+
+#### installing the optional dependencies
```
Comment (mathiascode) [inline doc/DEPENDENCIES.md:33]:
```suggestion
#### Installing the optional dependencies
```

File: doc/DEPENDENCIES.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+* On Redhat/Fedora based distributions:
+```
+sudo dnf install gsound gspell libappindicator-gtk3 libnotify python3-miniupnpc
```
Comment (mathiascode) [inline doc/DEPENDENCIES.md:36]: Can you add python3-miniupnpc to required dependencies? I forgot to change this.

File: doc/DEPENDENCIES.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+```
+* On Debian/Ubuntu based distributions:
+```
+sudo apt install gir1.2-appindicator3-0.1 gir1.2-gsound-1.0 gir1.2-gspell-1 gir1.2-notify-0.7 python3-miniupnpc
```
Comment (mathiascode) [inline doc/DEPENDENCIES.md:40]: Same here.

**Comment by hboetes** [comment]
Nifty interface for fixing conflicts. :-)

**Comment by hboetes** [comment]
@mathiascode I found a few more changes. :-}

**Comment by kiplingw** [comment]
@hboetes, a number of the checks are failing. Please look into this before I merge. You may simply need to rebase.

**Pull request merged**